### ISSUE TITLE
Phaser >= 3.50 Issue

### ISSUE THREAD

**Issue by asheraryam** [geckosio/phaser-on-nodejs#7]
Hello I just found this repo and I am using Phaser 3.50.

The readme mentions an error, what is it? I have no other option than to use phaser 3.50 so I would like to help fix it.

**Comment by yandeu** [comment]
The bug is in Phaser itselfe. The headless mode of Phaser does not work in 3.50.0.

Why don't you use an older version? 

You could also use v3.24.1 on the server and 3.50.0 in the client.

---

Two hours ago, v3.50.1 has been released. Maybe it works now. Do you want to check it?

**Comment by asheraryam** [comment]
@yandeu I started my project with 3.50 so I am not sure if I will have problems downgrading :shrug: 

I will do what you said and use 3.24 on the server since that will probably work, or maybe 3.50.1. Thanks for responding!

**Comment by yandeu** [comment]
Looks like it has been fixed and does now work in v3.51.0

https://github.com/photonstorm/phaser/issues/5468

**Comment by yandeu** [comment]
> Looks like it has been fixed and does now work in v3.51.0

_Nope, it does not work. Even v3.52.0 does not work yet._

**Comment by photonstorm** [comment]
There are no open issues about HEADLESS mode on the Phaser repo, so until someone creates one this is going to remain an issue (whatever it is)

**Comment by photonstorm** [comment]
Having said that, our example works fine in 3.51 and above: http://labs.phaser.io/view.html?src=src%5Cgame%20config%5Cheadless%20renderer.js so if you've managed to break it in a different way then it definitely needs reporting as an issue.

**Comment by yandeu** [comment]
The issue is not in the browser. It's on Node.js. 

Before v3.50 everything worked great. v3.50 created the issue photonstorm/phaser#5468. But since then, it does not seem to run on Node.js.

I admit I haven't investigated because I thought the huge update to 3.50 would bring some issues anyways and that this would be fixed automatically sooner or later. I was wrong :/

The code below works great in v3.24.1, but not in newer versions.

```json
{
  "scripts": {
    "start": "node index.js"
  },
  "dependencies": {
    "@geckos.io/phaser-on-nodejs": "^1.2.3",
    "phaser": "^3.24.1"
  }
}

```

```js
// index.js
require('@geckos.io/phaser-on-nodejs')
const Phaser = require('phaser')

class MainScene extends Phaser.Scene {
  create() {
    console.log('it works!')
  }
}

const config = {
  type: Phaser.HEADLESS,
  banner: false,
  audio: false,
  scene: [MainScene],
  physics: {
    default: 'arcade',
    arcade: {
      gravity: { y: 300 }
    }
  }
}

new Phaser.Game(config)
```

@photonstorm Do you know how to run Phaser.js on Node.js?

**Comment by sebastianfast** [comment]
I'm successfully running `phaser-on-nodejs` with the `phaser=3.54.0` now.

Perhaps unrelated, I noticed a problem where `game.postBoot()` was never getting called in the HEADLESS mode, see issue https://github.com/photonstorm/phaser/issues/5689. I created a pull request fixing this bug. You may want to use [my fork](https://github.com/sebastianfast/phaser) for now as long as my pull request isn't merged.

**Comment by yandeu** [comment]
@sebastianfast Thanks a lot!

_I personally wait for another release since I'm fine using an older version for now._

---

> I'm successfully running `phaser-on-nodejs` with the `phaser=3.54.0` now.

With your PR or the official 3.54.0 version?

**Comment by sebastianfast** [comment]
You're welcome. Made an edit to my original message, unrelated to your question

> > I'm successfully running `phaser-on-nodejs` with the `phaser=3.54.0` now.
> 
> With your PR or the official 3.54.0 version?

Yes and no. I got it working in the first place with the official 3.54.0. Then I stumbled above the bug mentioned in the issue photonstorm/phaser#5689 and issued a PR for the fix. If you do not need to use `game.postBoot()`, then you will most likely be fine. I think, however, that most people will need this.
